conda_recipes/Dockerfile

```diff
@@ -0,0 +1,68 @@
+FROM nvidia/cuda:8.0-cudnn6-devel-ubuntu14.04
```

Comment:
FMI, why do we rewrite a full new Dockerfile rather than build on top of what we have been using in CI?

Reply:
the one in CI doesn't have GPU but for building conda packages, I need to run python test which basically need GPU in all cases. Hence building the docker image which has GPUs. You can read how nvidia-docker works. I provided the readme instructions here for using and building this image.

Reply:
#resolved

Reply:
The docker image can't have GPUs, you can't download them :)

All our trusty docker images used to inherit from this ```FROM nvidia/cuda:8.0-cudnn6-devel-ubuntu14.04``` and all our xenial docker images used to inherit from this ```FROM nvidia/cuda:9.0-cudnn7-devel-ubuntu16.04``` but I now realize all that has changed drastically in the last stretch before the release and that we now use this [file](https://github.com/facebookresearch/TensorComprehensions/blob/master/docker/common/install_cuda.sh).

Still, when providing the Docker for conda_recipes you are using a very similar Docker base than what I wrote a few months back which got me confused. 

Does this mean we should uniformize what is in [here](https://github.com/facebookresearch/TensorComprehensions/tree/master/docker) and revert back to the old docker files like you did here?

I'll open a task specifically for this.

Reply:
The docker files will be moved from here to a separate repo https://github.com/prigoyal/tensorcomp-dockerfiles that I have setup for Jenkins setup. The question about whether the images should be tiered or not is something that I am discussing with @pietern and @ezyang already. Please open the issue in the above repo to centralize discussions properly.

>> Still, when providing the Docker for conda_recipes you are using a very similar Docker base than what I wrote a few months back which got me confused.

Not quite sure, the docker images follow quite different structure now and provide more safeguarding. 

>> FROM nvidia/cuda:8.0-cudnn6-devel-ubuntu14.04 

why did we inherit from those earlier? did you use nvidia-smi?

>> The docker image can't have GPUs, you can't download them :)

yeah, that's why we use `nvidia-docker` https://github.com/NVIDIA/nvidia-docker , looking at the diagram here will make things clear.

Reply:
#resolved

Reply:
> why did we inherit from those earlier? did you use nvidia-smi?

No since there is no cuda ***driver*** on the CPU-only CI machines.
However we have always had cross-compilation tests on CPU-only CI that were testing simple parts of the CUDA ***toolkit***. 

In particular you don't need a physical GPU to generate PTX and see whether compilation passes (e.g. NVRTC + CUB)...

Reply:
you can continue doing that, no need to start from NVIDIA images